### PR TITLE
chore: Use gopkg.in/yaml.v2 instead of v3

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

In PR https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1433 The usage of `gopkg.in/yaml.v3` was introduced. This added the usage of `gopkg.in/yaml.v2` and `gopkg.in/yaml.v3` depending on the package in the repository unnecessarily.

This PR removes the usage of `gopkg.in/yaml.v3` to unify the `gopkg.in/yaml` library version used

## Verification Steps

There are no references to `gopkg.in/yaml.v3` in the repository.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by miguelsorianod** [comment]
rebased

File: go.mod [hunk 1]
```diff
@@ -133,6 +131,7 @@ require (
 	google.golang.org/appengine v1.6.7 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```
Comment (JameelB) [inline go.mod:134]: This can be removed now no?

Comment (ziccardi) [reply go.mod:134]: I think it gets added automatically.

**Comment by ziccardi** [comment]
codecov upload failed: `Failed to properly upload: The process '/home/runner/work/_actions/codecov/codecov-action/v2/dist/codecov' failed with exit code 255`.

This happens quite often...

**Review by ziccardi — APPROVED** [review]
Thank you @miguelsorianod !

**Pull request merged**